pycbc/inference/models/tools.py

```diff
@@ -296,6 +296,26 @@ def snr_draw(self, snrs):
             # marginalizations.
             pass
 
+    def draw_ifos(self, snrs, snr_threshold=4.0):
```

Comment:
If the SNR is too low, then that detector is likely to just draw points that are not physical, hence it makes monte-carlo sampling inneficient. To avoid this, you require an actual peak to be present to include the detector when drawing samples. Keep in mind that if all detectors have no peak, it will fall back to just the prior values. (these are drawn ahead of time and ready to go in any case).